modules/services/views/src/main/java/au/com/shiftyjelly/pocketcasts/views/multiselect/MultiSelectHelper.kt

```diff
@@ -8,181 +8,100 @@ import androidx.fragment.app.FragmentManager
-import au.com.shiftyjelly.pocketcasts.preferences.Settings
-import au.com.shiftyjelly.pocketcasts.repositories.download.DownloadManager
-import au.com.shiftyjelly.pocketcasts.repositories.playback.PlaybackManager
-import au.com.shiftyjelly.pocketcasts.repositories.podcast.EpisodeManager
-import au.com.shiftyjelly.pocketcasts.repositories.podcast.PodcastManager
-import au.com.shiftyjelly.pocketcasts.repositories.podcast.UserEpisodeManager
-import au.com.shiftyjelly.pocketcasts.utils.combineLatest
-import au.com.shiftyjelly.pocketcasts.views.R
-import au.com.shiftyjelly.pocketcasts.views.dialog.ConfirmationDialog
+import au.com.shiftyjelly.pocketcasts.localization.R
 import au.com.shiftyjelly.pocketcasts.views.dialog.OptionsDialog
-import au.com.shiftyjelly.pocketcasts.views.helper.CloudDeleteHelper
-import au.com.shiftyjelly.pocketcasts.views.helper.DeleteState
 import com.google.android.material.snackbar.Snackbar
-import io.reactivex.BackpressureStrategy
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.launch
-import kotlinx.coroutines.withContext
-import timber.log.Timber
-import javax.inject.Inject
-import javax.inject.Singleton
 import kotlin.coroutines.CoroutineContext
-import kotlin.math.min
 import au.com.shiftyjelly.pocketcasts.images.R as IR
-import au.com.shiftyjelly.pocketcasts.localization.R as LR
-import au.com.shiftyjelly.pocketcasts.ui.R as UR
 
-private const val WARNING_LIMIT = 3
-@Singleton
-class MultiSelectHelper @Inject constructor(
-    val episodeManager: EpisodeManager,
-    val userEpisodeManager: UserEpisodeManager,
-    val podcastManager: PodcastManager,
-    val playbackManager: PlaybackManager,
-    val downloadManager: DownloadManager,
-    val settings: Settings,
-    private val episodeAnalytics: EpisodeAnalytics
-) :
-    CoroutineScope {
-    interface Listener {
+abstract class MultiSelectHelper<T> : CoroutineScope {
+    override val coroutineContext: CoroutineContext
+        get() = Dispatchers.Default
+
+    interface Listener<in T> {
         fun multiSelectSelectAll()
         fun multiSelectSelectNone()
-        fun multiSelectSelectAllUp(episode: BaseEpisode)
-        fun multiSelectSelectAllDown(episode: BaseEpisode)
+        fun multiSelectSelectAllUp(multiSelectable: T)
+        fun multiSelectSelectAllDown(multiSelectable: T)
     }
 
-    override val coroutineContext: CoroutineContext
-        get() = Dispatchers.Default
+    open lateinit var listener: Listener<T>
 
     private val _isMultiSelectingLive = MutableLiveData<Boolean>().apply { value = false }
     val isMultiSelectingLive: LiveData<Boolean> = _isMultiSelectingLive
 
+    protected val selectedList: MutableList<T> = mutableListOf()
+    protected val _selectedListLive = MutableLiveData<List<T>>().apply { value = listOf() }
+    val selectedListLive: LiveData<List<T>> = _selectedListLive
+    val selectedCount: LiveData<Int> = _selectedListLive.map { it.size }
```

Comment:
I haven't converted existing`LiveData` to `Flow` as it would further increase PR size. I've converted them to `Flow` in the `BookmarkViewModel`, mocking `LiveData` from `MultiSelectHelper` in the test class is not working well so I've commented out tests in [be86cb5](https://github.com/Automattic/pocket-casts-android/pull/1173/commits/be86cb507eb048a6008babf4fa9cbfd9ccf97bd9). 

I'll try to fix it in a follow up PR.